### PR TITLE
Unit Tests for `MultiMapUtils` and `HeaderInformation`

### PR DESCRIPTION
**Pull request by fussel178**
### Details 

Add Unit Tests for `MultiMapUtils` and `HeaderInformation`.

### Additional information 

This branch depends on the https://github.com/wuespace/telestion-core/tree/feat/api/improve-json-message branch because it needs the `MockMessage` implementation. Therefore, please first merge the mentioned branch into `main` and then this branch. :wink: 

### CLA

- [x] I have signed the individual contributor's license agreement **and sent** it to the board of the WüSpace e. V. organization.

### CODE REVIEW

**Review by pklaschka — APPROVED** [review]

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java [hunk 1]
```diff
@@ -1,141 +1,353 @@
...
-	 * Returns the simple class name of the subclass.
+	 * Constructs a {@link JsonMessage} from a buffer which contains an encoded JSON string.
 	 *
-	 * @return simple class name of subclass
+	 * @param json the buffer that contents contain the necessary information to construct
+	 *             the specified {@link JsonMessage}
+	 * @param type the class of the target {@link JsonMessage}
+	 * @param <T>  the type of the target {@link JsonMessage}
+	 * @return the decoded message
+	 * @throws DecodeException if the buffer contents does not contain the necessary information to successfully
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java:223]:
```suggestion
	 * @throws DecodeException if the buffer contents do not contain the necessary information to successfully
```

either "buffer contents" and "do", or "buffer content" and "does" :D

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java [hunk 2]
```diff
@@ -1,141 +1,353 @@
...
+	 * @throws EncodeException if the {@link JsonMessage} containing properties that cannot be represented
+	 *                         by JSON values
+	 * @see io.vertx.core.json.jackson.JacksonCodec#toString(Object, boolean)
+	 */
+	default String toJsonString(boolean pretty) throws EncodeException {
+		return Json.CODEC.toString(this, pretty);
+	}
+
+	/**
+	 * Like {@link #toJsonString(boolean)} but with space efficient JSON output.
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java:313]:
```suggestion
	 * Like {@link #toJsonString(boolean)}, but with space efficient JSON output.
```

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java [hunk 3]
```diff
@@ -1,141 +1,353 @@
...
+	 * @throws EncodeException if the {@link JsonMessage} containing properties that cannot be represented
+	 *                         by JSON values
+	 * @see io.vertx.core.json.jackson.JacksonCodec#toBuffer(Object, boolean)
+	 */
+	default Buffer toJsonBuffer(boolean pretty) throws EncodeException {
+		return Json.CODEC.toBuffer(this, pretty);
+	}
+
+	/**
+	 * Like {@link #toJsonBuffer(boolean)} but with space efficient JSON output.
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java:333]:
```suggestion
	 * Like {@link #toJsonBuffer(boolean)}, but with space efficient JSON output.
```

**Review by cb0s — DISMISSED** [review]
Noice! 🎉

**Pull request merged**